packages/app/src/app/pages/common/Modals/NewTeamModal/TeamMembers.tsx

```diff
@@ -35,10 +36,35 @@ export const TeamMembers: React.FC<{ onComplete: () => void }> = ({
   onComplete,
 }) => {
   const { activeTeamInfo } = useAppState();
+  const actions = useActions();
   const { gql } = useEffects();
+  const { copyToClipboard } = useEffects().browser;
   const [addressesString, setAddressesString] = useState<string>();
   const [invalidEmails, setInvalidEmails] = useState<string[]>();
   const [inviteError, setInviteError] = useState<string>();
+  const [linkCopied, setLinkCopied] = React.useState(false);
+
+  const copyLinkTimeoutRef = React.useRef<number>();
+
+  const inviteLink = teamInviteLink(activeTeamInfo.inviteToken);
+
+  const copyTeamInviteLink = () => {
+    copyToClipboard(inviteLink);
+    setLinkCopied(true);
+
+    if (copyLinkTimeoutRef.current) {
+      window.clearTimeout(copyLinkTimeoutRef.current);
+    }
+
+    actions.track({
+      name: 'Dashboard - Copied Team Invite URL',
+      data: { place: 'modal', inviteLink },
+    });
+
+    copyLinkTimeoutRef.current = window.setTimeout(() => {
+      setLinkCopied(false);
+    }, 1500);
```

Comment:
Nitpick, but we can get rid of `window` for `*Timeout` functions if we want. We do have to type the timeout different in that case though.

```suggestion
  const copyLinkTimeoutRef = React.useRef<ReturnType<typeof setTimeout>>();

  const inviteLink = teamInviteLink(activeTeamInfo.inviteToken);

  const copyTeamInviteLink = () => {
    copyToClipboard(inviteLink);
    setLinkCopied(true);

    if (copyLinkTimeoutRef.current) {
      clearTimeout(copyLinkTimeoutRef.current);
    }

    actions.track({
      name: 'Dashboard - Copied Team Invite URL',
      data: { place: 'modal', inviteLink },
    });

    copyLinkTimeoutRef.current = setTimeout(() => {
      setLinkCopied(false);
    }, 1500);
```

Reply:
Nice! Thanks for the tip!